sdk/typescript/type_guards.mjs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
I saw a comment to move this to a node script, figured I'd do it while I was here.